pint_server/app.py

```diff
@@ -425,11 +439,28 @@ def get_provider_servers(provider):
             for server in servers]
 
 
+def trim_images_payload(images):
+    payload_size = jsonify(images=images).content_length
+
+    if payload_size >  MAX_PAYLOAD_SIZE:
+        # NOTE: assuming the size of the entries are evenly distributed, we
+        # determine the percentage of entries to trim from the end of the list,
+        # as the list is sorted in decending order by publishedon date, by
+        # calculating the percentage over the maximum payload size. Then
+        # trim the same percentage off the list, rounding up to be safe.
+        trim_size  = math.ceil(
+            ((payload_size - MAX_PAYLOAD_SIZE) / payload_size) * len(images))
+        images = images[:-trim_size]
```

Comment:
This will create the problem that we might send partial data for a given image. For example we might send data for `suse-sles-12-sp1-v20170630-hvm...` for only some regions. Logic that is strictly based on math is not sufficient. We'll have to find the first image name that got cut and then make sure no images with that name are in the data that we would send.

Reply:
Make sense. Will do.